### PR TITLE
Save input value as IChatTransferData for restoring after transfer

### LINKED ISSUE #187657: Open Session in Editor loses input editor contents
1. Open chat in sidebar
2. Type something into input
3. Open Session in Editor
4. :bug: input is cleared

### PR DESCRIPTION
**Pull request by bhavyaus**
This fixes restoring chat input on transferring to a new workspace.

### CODE REVIEW

**Comment by roblourens** [comment]
I don't like this approach because `IChatModel` seems like the wrong place for this data, and that interface is read-only for all other content. I'd have to think about a better way to transfer some view state along with the session data. The idea is to keep the same input in the input box after you click the button to open the new project, in case I started typing a question? That's ok, it doesn't sound like a common scenario though.

**Comment by bhavyaus** [comment]
> The idea is to keep the same input in the input box after you click the button to open the new project, in case I started typing a question? That's ok, it doesn't sound like a common scenario though.

Yes. After we've transferred the session over to a new workspace, we'd like to preserve the `/createWorkspace` in the input. 
The other scenario is moving from the panel chat to the editor. 

Currently we store and restore the input state in the chatViewPane at a WORKSPACE level. I'll check to see there's a change I can make there.

File: src/vs/workbench/api/browser/mainThreadChat.ts [hunk 1]
```diff
@@ -54,7 +54,9 @@ export class MainThreadChat extends Disposable implements MainThreadChatShape {
 	}
 
 	$transferChatSession(sessionId: number, toWorkspace: UriComponents): void {
-		this._chatService.transferChatSession(sessionId, URI.revive(toWorkspace));
+		const widget = this._chatWidgetService.lastFocusedWidget;
```
Comment (roblourens) [inline src/vs/workbench/api/browser/mainThreadChat.ts:57]: You'll want to find the widget that has the right sessionId, you could add a method to the ChatWidgetService for this

File: src/vs/workbench/contrib/chat/common/chatService.ts [hunk 2]
```diff
@@ -194,6 +194,7 @@ export const IChatService = createDecorator<IChatService>('IChatService');
 export interface IChatService {
 	_serviceBrand: undefined;
 	transferredSessionId: string | undefined;
```
Comment (roblourens) [inline src/vs/workbench/contrib/chat/common/chatService.ts:196]: These should be tied together in one object

**Review by roblourens — APPROVED** [review]

**Pull request merged**